### PR TITLE
[camera] Android: Improve image streaming by creating a request suita…

### PR DESCRIPTION
**Pull request by PerLycke**
…ble for video recording instead of still image capture.

## Description

Existing behaviour creates a request which prioritizes image quality over frame rate and results in a choppy and / or freezing streaming session on many devices. Creating a request suitable for video recording means that a stable frame rate is used, and post-processing is set for recording quality. It noticeably improves streaming images on a variety of devices.

## Related Issues

Fixes flutter/flutter#43566

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java [hunk 1]
```diff
@@ -426,7 +426,7 @@ public void startPreview() throws CameraAccessException {
 
   public void startPreviewWithImageStream(EventChannel imageStreamChannel)
       throws CameraAccessException {
-    createCaptureSession(CameraDevice.TEMPLATE_STILL_CAPTURE, imageStreamReader.getSurface());
+    createCaptureSession(CameraDevice.TEMPLATE_RECORD, imageStreamReader.getSurface());
```
[the remark this answers is not in the document]
Comment (PerLycke) [reply packages/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java]: I've tested with both but using TEMPLATE_PREVIEW breaks trying to detect objects in a CameraImage using the ml kit on my test devices. With TEMPLATE_RECORD it both fixes the streaming issues and keeps the functionality intact. @mehmetf might have more insight in this as I know he looked in to improving streaming.

**Comment by PerLycke** [comment]
@mklim I've added a basic test for streaming on Android. If I got something wrong, or if more sophisticated tests are needed, I will need to hand this over to someone in the Flutter team.

File: packages/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (PerLycke) [reply packages/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java]: Thanks Mehmet, I hope some of those improvements can find its way to this plugin later. As of now, and regarding this PR, changing TEMPLATE_STILL_CAPTURE to TEMPLATE_RECORD singlehandedly fixes streaming on several devices. There's not much documentation around this in particular, but seems needed

**Comment by tigrenok00** [comment]
Hi, did this make into latest stable version? We're still experiencing performance issues with CameraPreview freezing while streaming images with High resolution, and I thought this might fix this.
Thanks

**Comment by PerLycke** [comment]
@tigrenok00 It's included yes, but it won't enable streaming images on high resolution on all devices. It fixes so that you can use recommended resolution on all devices (afaik) without notable issues. Before this patch camera plugin was unusable on a variety of devices, even on lower resolutions.

**Comment by datovarh** [comment]
Hi, can you help me ? with this https://github.com/flutter/flutter/issues/60975

**Pull request merged**